### DISCUSSION TITLE
Getting python chucks to run for the first time -- Error: (mach-o file, but is an incompatible architecture (have (x86_64), need (arm64e)))

### DISCUSSION THREAD

**Discussion by kellygarvey** [quarto-dev/quarto-cli#7051] [Q&A] [answered]
### Description

I am looking forward to starting to use the python chunks in quarto but I'm having trouble with this error message: (mach-o file, but is an incompatible architecture (have (x86_64), need (arm64e))). 

I copied and tried to run the sample code from https://quarto.org/docs/computations/python.html but I get: 
> reticulate::repl_python()
Error: /Users/kellyanngarvey/opt/anaconda3/lib/libpython3.9.dylib - dlopen(/Users/kellyanngarvey/opt/anaconda3/lib/libpython3.9.dylib, 0x000A): tried: '/Users/kellyanngarvey/opt/anaconda3/lib/libpython3.9.dylib' (mach-o file, but is an incompatible architecture (have (x86_64), need (arm64e)))

I have anaconda installed and I just upated it. My version of R is also up to date. 

I greatly appreciate any help on this!
Thanks so much, 
Kelly

**Comment by mcanouil** [comment, marked as answer]
It seems you have a conflict of platform architecture, namely arm64 versus x86-84 probably resulting from installing the wrong tools.

What's your operating system and platform?
Did you install the proper tools in the same architecture as your operating system?


From the message, I am guessing you have macOS with an Apple Silicon chip which is arm64 and somehow your anaconda environment is using x86-64 while it should be arm64.

**Reply by kellygarvey** [reply]
I'm on a Mac Air version 12.6.

I installed re intalled the anaconda with the arm64. I didn't realize there were two types of Mac downloads. 

I re ran the code chunks and now it is having trouble reading in the packages: >>> import numpy as np
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/reticulate/python/rpytools/loader.py", line 119, in _find_and_load_hook
    return _run_hook(name, _hook)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/reticulate/python/rpytools/loader.py", line 93, in _run_hook
    module = hook()
             ^^^^^^
  File "/Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/reticulate/python/rpytools/loader.py", line 117, in _hook
    return _find_and_load(name, import_)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'numpy'

**Reply by mcanouil** [reply]
You have to install the modules used in the code.
This step is (or should be) described in the documentation next to the example.

```bash
pip3 install jupyter numpy matplotlib
```

Or 

```bash
python3 -m pip install jupyter numpy matplotlib
```

Note that these commands need to be executed in your activated Conda environment.

**Reply by kellygarvey** [reply]
Great! This worked, I can run the chunk. 

For any other packages I have to install, can I use "pip3 install jupyter [package name here]"? Is that the correct way to think about this? 

Thank you SO MUCH for your quick response and help!!

**Reply by mcanouil** [reply]
`jupyter` is already a module/library, see <https://quarto.org/docs/computations/python.html#installation>.
